core/server/worker/src/main/java/alluxio/worker/block/management/tier/SwapRestoreTask.java

```diff
@@ -141,12 +141,15 @@ public BlockManagementTaskResult run() {
         long blockId = tierIterator.next();
         try {
           BlockMeta nextBlockFromTier = mEvictorView.getBlockMeta(blockId);
-          moveOutBytes -= nextBlockFromTier.getBlockSize();
-          if (lastTier) {
-            blocksToRemove.add(nextBlockFromTier.getBlockId());
-          } else {
-            blocksToTransfer.add(BlockTransferInfo.createMove(nextBlockFromTier.getBlockLocation(),
-                nextBlockFromTier.getBlockId(), destLocation));
+          if (nextBlockFromTier != null) {
```

Comment:
can you check if line 189 too? May that create NPE too?

Reply:
Right. Fixed that too.